src/pal/src/synchmgr/wait.cpp

```diff
@@ -531,15 +531,19 @@ DWORD CorUnix::InternalWaitForMultipleObjectsEx(
     iSignaledObjIndex = -1;
     for (i=0;i<(int)nCount;i++)
     {
-        bool fValue;
-        palErr = ppISyncWaitCtrlrs[i]->CanThreadWaitWithoutBlocking(&fValue, &fAbandoned);
+        bool fValue, fWaitObjectAbandoned = false;
+        palErr = ppISyncWaitCtrlrs[i]->CanThreadWaitWithoutBlocking(&fValue, &fWaitObjectAbandoned);
         if (NO_ERROR != palErr)
         {
             ERROR("ISynchWaitController::CanThreadWaitWithoutBlocking() failed for "
                   "%d-th object [handle=%p error=%u]\n", i, lpHandles[i], palErr);
             pThread->SetLastError(ERROR_INTERNAL_ERROR);
             goto WFMOExIntReleaseControllers;            
         }
+        if (fWaitObjectAbandoned)
```

Comment:
A micro-nit: you could use `fAbandoned = fAbandoned | fWaitObjectAbandoned` instead of the condition.

Reply:
I usually prefer `if` over `|=` if just for what I like to see, but is this better in some way?

Reply:
I would expect the compiler to generate a branchless code and less branching is always better. In this place however, it doesn't really matter, so it was just my personal preference.
Surprisingly, using the compiler explorer, I can see that the variant with "if" generates better code with optimizations enabled.
So please keep it as it is.

Reply:
Ok will do. When the condition (that the wait object is abandoned) is most often false, even with ideal code would branchless code be better than the branch?

Reply:
I suppose if `fAbandoned` is enregistered then branchless code could be better, and if it's not enregistered then branchless code could be worse. Probably wouldn't make much difference either way due to the predictable branch and likely prefetchable/cached memory.

Reply:
If that's true I figure the compiler should be able to decide better on which code to generate based on whether what is enregistered, and from a source code perspective, there should be no real difference between the two. In an ideal perspective anyway :)